pipeline/backprocess.go

```diff
@@ -3,118 +3,115 @@ package pipeline
 import (
 	"context"
 	"fmt"
+	pbsubstreams "github.com/streamingfast/substreams/pb/sf/substreams/v1"
 
 	"github.com/streamingfast/substreams"
 	"github.com/streamingfast/substreams/orchestrator"
 	"github.com/streamingfast/substreams/state"
-	"go.opentelemetry.io/otel/codes"
 	"go.uber.org/zap"
 )
 
 func (p *Pipeline) backProcessStores(
 	ctx context.Context,
 	workerPool *orchestrator.WorkerPool,
-	initialStoreMap map[string]*state.Store,
-) (
-	map[string]*state.Store,
-	error,
-) {
-	ctx, span := p.tracer.Start(p.context, "back_processing")
-	defer span.End()
-
-	logger := p.logger.Named("back_process")
+	storeModules []*pbsubstreams.Module,
+) (out map[string]*state.Store, err error) {
+	p.reqCtx.StartSpan("back_processing")
+	successMessage := "completed"
+	defer func() {
+		if err != nil {
+			p.reqCtx.SetErr(err)
+		} else {
+			p.reqCtx.SetOk(successMessage)
+		}
+		p.reqCtx.EndSpan()
+	}()
```

Comment:
I think this could be replaced with something much leaner:
```
defer p.reqCtx.StartSpan("back_processing").EndSpan()
defer p.reqCtx.SetReturn(&err)
```
and have `SetReturn` distinguish between a `nil` (and set `completed` by default, or I suggest `ok` as a simple affirmative, shorter, less data loaded in the tracer APIs).

One could even think of a return of a `SuccessMessage("my message")` that would be altered into a `nil` in the `*err` within `SetReturn`.

My suggestion though? Drop the `SetOk()` message altogether, unless we feel we _really_ need the distinguished success messages.